Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -122,31 +124,33 @@ void DVDInterface::DoState(PointerWrap& p)
   m_adpcm_decoder.DoState(p);
 }
 
-size_t DVDInterface::ProcessDTKSamples(std::vector<s16>* temp_pcm,
+size_t DVDInterface::ProcessDTKSamples(s16* target_samples, size_t block_count,
                                        const std::vector<u8>& audio_data)
 {
-  size_t samples_processed = 0;
-  size_t bytes_processed = 0;
-  while (samples_processed < temp_pcm->size() / 2 && bytes_processed < audio_data.size())
+  size_t available_block_count =
```

Comment:
Nit: `available_block_count` sounds to me like something derived from the input, when really here it's derived from the size of the provided output buffer. `requested_block_count` maybe? Or feel free to nak.

Reply:
Well it's both, `audio_data.size()` is the input, and in practice that will be the limiter instead of the `block_count`... I think I'll change the `block_count` parameter to be a more specific `target_block_count` and then the `available_block_count` to `block_count_to_process`.